File: src/vscodeextension/src/test/testfiles/typescript/xx_manual_tests.ts


let a = 4;
let ab = 4;
let ac = 4;
let ad = 4;
let ae = 4;

function ccc() {
  return 5;
}

function ddd() {
  let a = new Test();
  exAsyncFunc();
  return ccc ();
}

export function exFunc()
{
  exAsyncFunc ();
  return ccc();
}

export async function exAsyncFunc()
{
  exFunc() ;
  return ccc();
}



class Test{

  aaa(){                 // returns: vscode.LocationLink

  }

  bbb(){                 // returns: vscode.LocationLink
    return this.aaa ();  // returns: vscode.LocationLink
  }

  async ccc(){
    return this.aaa ();
  }

}

class Parent {
  public foo()  { return "Parent::foo()"; }
  public foo2() { return 11; }
}

class ChildA extends Parent {
  public foo() { return "ChildA::foo()"; }
}

class ChildB extends Parent {
  public foo() { return "ChildB::foo()"; }
}

function childparenttest(){
  let test: Parent = new ChildA();
  test.foo();
}

File: src/vscodeextension/src/test/testfiles/python/xx_manual_tests.py
def aaa():
  return 5

def bbb():
  return aaa()

# a comment

def bbb2():
  a = Test()
  return aaa()


a = 14

class Test:

  def aaa(self):
    return 5


  def bbb(self):
    return self.aaa ()

  def ccc(self):
    return self.aaa ()



class Parent :
   def foo():
      return "parent::foo()"
   def foo2():
      return 11


class ChildA (Parent):
  def foo():
    return "ChildA::foo()"
  def foo2(param1,
           param2):
    return 3

class ChildB (Parent):
  def foo():
   return "ChildB::foo()"


def childparenttest():
  """explaining the function"""
  # short comment

  test = None
  if (input() > 1):
    test = ChildA()
  else:
    test = ChildB()

  def asdf():
    pass

  test.foo()


def breaking_parameters (param1,
                         param2,
                         param3):
  pass


def long_long_long_method (param1, param2,
                           param3, param4,
                           param5, param6):
  pass
  pass
  pass


def long_long_long_method2 (param1, param2,
                            param3, param4,
                            param5, param6):
  a = 5 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1
  pass
  pass

